### ISSUE TITLE
Error met when running BERT training on multiple nodes

### ISSUE THREAD

**Issue by Eugene-Mark** [IntelAI/models#94]
**Issue:**
/usr/bin/python3 common/tensorflow/run_tf_benchmark.py --framework=tensorflow --
use-case=language_modeling --model-name=bert_large --precision=fp32 --mode=train
ing --benchmark-dir=/dl/intel_train/models/benchmarks --intelai-models=/dl/intel
_train/models/benchmarks/../models/language_modeling/tensorflow/bert_large --num
-cores=-1 --batch-size=32 --socket-id=-1 --output-dir=/dl/intel_train/glue-outpu
t --num-train-steps=1 --benchmark-only --num-intra-threads=10 --disable-tcmalloc
=True --train-option=Classifier --init-checkpoint=/dl/intel_train/ckpt-bert-base
/uncased_L-12_H-768_A-12/bert_model.ckpt --task-name=MRPC --vocab-file=/dl/intel
_train/ckpt-bert-base/uncased_L-12_H-768_A-12/vocab.txt --config-file=/dl/intel_
train/ckpt-bert-base/uncased_L-12_H-768_A-12/bert_config.json --do-train=true --
num-train-epochs=30 --learning-rate=2e-5 --max-seq-length=128 --do-eval=true --d
ata-dir=/dl/intel_train/glue/results4/MRPC --do-lower-case=True --experimental-g
elu=False --optimized-softmax=True
<2> is invalid
libnuma: Warning: node argument 2 is out of range

usage: numactl [--all | -a] [--interleave= | -i <nodes>] [--preferred= | -p <nod
e>]

**Env:**
OS: Fedora release 29 (Twenty Nine)   
Whether container: bare metal  
Model: bert base  
Reference guideline:    
https://github.com/IntelAI/models/tree/master/benchmarks/language_modeling/tensorflow/bert_large/training/fp32  
Number_of_nodes: 2
Socket_of_nodes: 2

**The code where throws the issue:**
Start.sh -> run_model()->eval ${CMD} 2>&1 | tee ${LOGFILE}

Note, the ${CMD} after preprocessing is:
_/usr/bin/python3 common/tensorflow/run_tf_benchmark.py --f
ramework=tensorflow --use-case=language_modeling --model-name=bert_large --preci
sion=fp32 --mode=training --benchmark-dir=/dl/intel_train/models/benchmarks --in
telai-models=/dl/intel_train/models/benchmarks/../models/language_modeling/tenso
rflow/bert_large --num-cores=-1 --batch-size=32 --socket-id=-1 --output-dir=/dl/
intel_train/glue-output --num-train-steps=1   --benchmark-only    --disable-tcma
lloc=True --train-option=Classifier --init-checkpoint=/dl/intel_train/ckpt-bert-
base/uncased_L-12_H-768_A-12/bert_model.ckpt --task-name=MRPC --vocab-file=/dl/i
ntel_train/ckpt-bert-base/uncased_L-12_H-768_A-12/vocab.txt --config-file=/dl/in
tel_train/ckpt-bert-base/uncased_L-12_H-768_A-12/bert_config.json --do-train=tru
e --num-train-epochs=30 --learning-rate=2e-5 --max-seq-length=128 --do-eval=true
--data-dir=/dl/intel_train/glue/results4/MRPC --do-lower-case=True --experiment
al-gelu=False --optimized-softmax=True_                                          

**The step to launch the training:**
•	Git clone intel model on each node with same directory
•	Prepare glue data on each node with same directory
•	Prepare bert model on each node with same directory
•	Make sure for each node, distributed mode on two sockets is workable per the guide. The train result can be generated on each node.
•	To support multiple instances, there is no guide for Bert so I’m referring the resnet related training https://github.com/IntelAI/models/blob/master/benchmarks/image_recognition/tensorflow/resnet50v1_5/training/fp32/Advanced.md
python launch_benchmark.py \
  --verbose \
  --model-name resnet50v1_5 \
  --precision fp32 \
  --mode training \
  --framework tensorflow \
  --noinstall \
  --checkpoint ${OUTPUT_DIR} \
  --data-location ${DATASET_DIR} \
  --mpi_hostnames 'host1,host2' \
  --mpi_num_processes 4 2>&1
•	I just modify the fp32_classifier_training.sh directly in launching entry, something like below:
source "${MODEL_DIR}/quickstart/common/utils.sh"
_command python3 ${MODEL_DIR}/benchmarks/launch_benchmark.py \
  --model-name=bert_large \
  --precision=fp32 \
  --mode=training \
  --framework=tensorflow \
  --batch-size=32 \
  ${mpi_num_proc_arg} \
  --mpi_hostnames='vsr140,vsr143' \
  --output-dir=$OUTPUT_DIR \
  $@ \
  -- train-option=Classifier \
  task-name=MRPC \
  do-train=true \
  do-eval=true \
  data-dir=$DATASET_DIR/MRPC \
  vocab-file=$CHECKPOINT_DIR/vocab.txt \
  config-file=$CHECKPOINT_DIR/bert_config.json \
  init-checkpoint=$CHECKPOINT_DIR/bert_model.ckpt \
  max-seq-length=128 \
  learning-rate=2e-5 \
  num-train-epochs=30 \
  optimized_softmax=True \
  experimental_gelu=False \
  do-lower-case=True \

**what we have tried**
We found some numactl related code under common/base_model_init.py so that we added some print out there, but found the error is threw before the numactl is invoked.

**Comment by sramakintel** [comment]
@Eugene-Mark do you still need assistance on this issue?

**Comment by Eugene-Mark** [comment]
@sramakintel, thanks for tracking the issue. I don't need support about it anymore, FYI.
